### PR TITLE
Allows `./gradlew clean build` to run successfully.

### LINKED ISSUE #352: `gradle build` fails
One can say that master branch is currently broken, because just running `./gradlew clean build` will result in build failure. This is certainly confusing to new contributors and leaves very bad impression on the project as a whole. 

I can take care of this problem, please assign this task to me.

### PR DESCRIPTION
**Pull request by iNikem**
Fixes #352.

For now the unsolved problem is code coverage verification. Several modules have significantly lower coverage than required by build. Thus this verification cannot be enabled until tests are improved.

### CODE REVIEW

**Comment by trask** [comment]
Hey @iNikem, reviewing this made me remember that we used to have code coverage verification, because i remember having to tweak the percentages a couple of times. It looks like I accidentally dropped the verification from CI a while back (while trying to save CI cost), and after that unsurprisingly the verification started failing. I submitted #357 to bring back the CI verification, and fix the regressions that crept in during its absence.

File: agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/jdbc/DBInfo.java [hunk 1]
```diff
@@ -30,4 +30,6 @@
   private final String db;
   private final String host;
   private final Integer port;
+
+  public static class Builder {}
```
Comment (iNikem) [inline agent-bootstrap/src/main/java/io/opentelemetry/auto/bootstrap/instrumentation/jdbc/DBInfo.java:34]: Without this explicit declaration javadoc generation for `JDBCConnectionUrlParser` fails with 
```
error: cannot find symbol
  abstract DBInfo.Builder doParse(String jdbcUrl, final DBInfo.Builder builder);
```

File: agent-bootstrap/src/main/java/io/opentelemetry/auto/config/Config.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
  * <p>
  *
  * <p>System properties are {@link Config#PREFIX}'ed. Environment variables are the same as the
- * system property, but uppercased with '.' -> '_'.
+ * system property, but uppercased and '.' is replaced with '_'.
```
Comment (iNikem) [inline agent-bootstrap/src/main/java/io/opentelemetry/auto/config/Config.java:45]: Apparently symbol `>` is not allowed in javadoc

File: gradle/java.gradle [hunk 3]
```diff
@@ -181,6 +181,7 @@ javadoc {
     setAuthor true
 
     links "https://docs.oracle.com/javase/8/docs/api/"
+    source = 8
```
Comment (iNikem) [inline gradle/java.gradle:184]: Without this javadoc fails on java 11 due to https://bugs.openjdk.java.net/browse/JDK-8212233

**Comment by iNikem** [comment]
> Would you mind commenting on each of the changes and why they were needed?

@tylerbenson Done

**Comment by tylerbenson** [comment]
@iNikem thanks!

**Comment by tylerbenson** [comment]
You may want to update CI to somehow exercise this to ensure it doesn't break again.

File: gradle/jacoco.gradle [hunk 4]
```diff
@@ -60,5 +60,5 @@ afterEvaluate {
   }
 
   jacocoTestCoverageVerification.dependsOn jacocoTestReport
-  check.dependsOn jacocoTestCoverageVerification
+//  check.dependsOn jacocoTestCoverageVerification
```
Comment (trask) [inline gradle/jacoco.gradle:63]: is this still needed now that jacoco verification is fixed in #357 (i just merged master into this PR)

**Comment by iNikem** [comment]
@trask @tylerbenson what do you think about changing `test_8` CI job to just run `build` task? And probably it will make sense to rename current `build` CI job to `assemble`

**Comment by iNikem** [comment]
Make a smaller change: `check` CI job will run `build`

File: gradle/jacoco.gradle (hunk 4 above)
Comment (iNikem) [reply gradle/jacoco.gradle]: No, reverted.

**Pull request merged**